src/pipx/paths.py

```diff
@@ -0,0 +1,86 @@
+import os
+from pathlib import Path
+from typing import Optional, Union
+
+from platformdirs import user_cache_path, user_data_path, user_log_path
+
+DEFAULT_PIPX_HOME = user_data_path("pipx")
+FALLBACK_PIPX_HOME = Path.home() / ".local/pipx"
+DEFAULT_PIPX_BIN_DIR = Path.home() / ".local/bin"
+DEFAULT_PIPX_MAN_DIR = Path.home() / ".local/share/man"
+DEFAULT_PIPX_GLOBAL_HOME = "/opt/pipx"
+DEFAULT_PIPX_GLOBAL_BIN_DIR = "/usr/local/bin"
+DEFAULT_PIPX_GLOBAL_MAN_DIR = "/usr/local/share/man"
+
+
+def get_expanded_environ(env_name: str) -> Optional[Path]:
+    val = os.environ.get(env_name)
+    if val is not None:
+        return Path(val).expanduser().resolve()
+    return val
+
+
+class _PathContext:
+    _base_home: Optional[Union[Path, str]] = get_expanded_environ("PIPX_HOME")
+    _base_bin: Optional[Union[Path, str]] = get_expanded_environ("PIPX_BIN_DIR")
+    _base_man: Optional[Union[Path, str]] = get_expanded_environ("PIPX_MAN_DIR")
+    _base_shared_libs: Optional[Union[Path, str]] = get_expanded_environ("PIPX_SHARED_LIBS")
+    _fallback_home: Path = Path.home() / ".local/pipx"
+    _in_home: bool = _base_home is not None or _fallback_home.exists()
+    log_file: Optional[Path] = None
+
+    @property
+    def venvs(self) -> Path:
+        return self.home / "venvs"
+
+    @property
+    def logs(self) -> Path:
+        if self._in_home:
+            return self.home / "logs"
+        return user_log_path("pipx")
+
+    @property
+    def trash(self) -> Path:
+        if self._in_home:
+            return self.home / ".trash"
+        return self.home / "trash"
```

Comment:
Could you explain why this is special-cased for the default pipx home? Since the `venvs` path isn't.

```suggestion
        return self.home / "trash"
```

Reply:
That's just an implementation of how the things are working on main: https://github.com/pypa/pipx/blob/0cfcdc1781d6bc366e7f72af5b3c59513d9ec118/src/pipx/constants.py#L28-L36
In this PR it probably makes sense to stay consistent with the old logic and change it in another PR if needed.